static/js/copy_and_paste.js

```diff
@@ -318,6 +331,23 @@ export function paste_handler(event) {
     }
 
     if (clipboardData.getData) {
+        const paste_text = clipboardData.getData("text");
+        if (paste_text && isValidHttpUrl(paste_text) && document.getSelection().type === "Range") {
+            event.preventDefault();
+            event.stopPropagation();
+
+            const textarea = $(this);
+            const prefix = "[";
+            const suffix = "](" + paste_text + ")";
+            const range = textarea.range();
+
+            if (!document.execCommand("insertText", false, prefix + range.text + suffix)) {
+                textarea.range(range.start, range.end).range(prefix + range.text + suffix);
+            }
```

Comment:
I think this should be using the common functions in `compose_ui.js` -- we try to have all logic inserting content into the compose box content use that, so that we can fix bugs in one place.  Which reminds me that we need tests; I think it's likely we'll want to build on https://github.com/zulip/zulip/pull/17864 but it's possible the existing testing system support this.

Reply:
Actually I used the logic from https://github.com/zulip/zulip/blob/e231a03eff6cce62243284ebcf7f06f798715a78/static/js/compose.js#L839 in `compose.js`
I will see, how to convert this into a function and add tests for that.

Reply:
I think, in this I will also refractor and move this into `compose_ui.js`
So, that they both remains in the same place.

Reply:
I have updated the PR. Now tests are left.